recipes/mp-units/0.7.0/conanfile.py

```diff
@@ -0,0 +1,84 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import Version, check_min_cppstd
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+class MPUnitsConan(ConanFile):
+    name = "mp-units"
+    homepage = "https://github.com/mpusz/units"
+    description = "Physical Units library for C++"
+    topics = ("units", "dimensions", "quantities", "dimensional-analysis", "physical-quantities", "physical-units", "system-of-units", "cpp23", "cpp20", "library", "quantity-manipulation")
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake_find_package_multi"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        compiler = self.settings.compiler
+        self.requires("fmt/7.1.3")
+        self.requires("gsl-lite/0.38.0")
+        if compiler == "clang" and compiler.libcxx == "libc++":
+            self.requires("range-v3/0.11.0")
+
+    def validate(self):
+        compiler = self.settings.compiler
+        version = Version(self.settings.compiler.version)
+        if compiler == "gcc":
+            if version < "10.0":
+                raise ConanInvalidConfiguration("mp-units requires at least g++-10")
+        elif compiler == "clang":
+            if version < "12":
+                raise ConanInvalidConfiguration("mp-units requires at least clang++-12")
+        elif compiler == "Visual Studio":
+            if version < "16":
+                raise ConanInvalidConfiguration("mp-units requires at least Visual Studio 16.9")
+        else:
+            raise ConanInvalidConfiguration("Unsupported compiler")
+        if compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "20")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        cmake = CMake(self)
+        cmake.configure(source_folder=os.path.join(self._source_subfolder, "src"))
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        compiler = self.settings.compiler
+
+        # core
+        self.cpp_info.components["core"].requires = ["gsl-lite::gsl-lite"]
```

Comment:
The components can not have generic names since it produces can conflict with other package names.

Reply:
How is that?
Won't they get namespaced with `mp-units::`?

Reply:
From the generatored `mp-unitsTargets.cmake`:

```cmake
if(NOT TARGET mp-units::core)
    add_library(mp-units::core INTERFACE IMPORTED)
endif()

if(NOT TARGET mp-units::isq)
    add_library(mp-units::isq INTERFACE IMPORTED)
endif()

if(NOT TARGET mp-units::si)
    add_library(mp-units::si INTERFACE IMPORTED)
endif()

# ...
```

Reply:
if a consumer uses the pkg_config generator, then it will create `core.pc`, which definitely has a high conflict potential with other recipes.

Reply:
Aha thanks.
So I guess it's better to pretend them with e.g. an underscore.

Reply:
But those are the official names used by the library. Do I really have to provide:

```python
        self.cpp_info.components["_xxx"].names["cmake_find_package"] = "xxx"
        self.cpp_info.components["_xxx"].names["cmake_find_package_multi"] = "xxx"
```

for each and every component in the list? I personally think that sucks 😢 

Also how `_core` is better than `core` if everyone will be forced to add `_` prefix to their generic names? 😉

Reply:
Maybe this is something for the Conan repository: _how to handle file collisions in pkgconfig world_ due to the usage of components because of CMake consumers.

We are not going to modify all the recipes to add those changes 😄

Reply:
that's probably not just a pkg-config, but any generator in general. as long as two projects define the same component such as `core`, files like `core.pc` or `core-config.cmake` will collide, at least if we use the same directory for all dependencies.
it's probably better to isolate them in their own directories to avoid such collisions.

Reply:
> it's probably better to isolate them in their own directories to avoid such collisions

How can I achieve that?

Reply:
conan does not support this (currently).
This would require adding paths to the `PKG_CONFIG_PATH` environment variable.
I don't know whether the search order is documented.
Putting the generated `.pc` files in different folders or overwriting them, might result in the same behavior.